sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosClientTelemetryConfig.java

```diff
@@ -0,0 +1,348 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.azure.core.http.ProxyOptions;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.implementation.Configs;
+import com.azure.cosmos.implementation.ImplementationBridgeHelpers;
+import com.azure.cosmos.implementation.Strings;
+import com.azure.cosmos.implementation.Utils;
+import com.azure.cosmos.implementation.apachecommons.lang.StringUtils;
+import com.azure.cosmos.implementation.clienttelemetry.TagName;
+import com.azure.cosmos.util.Beta;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Stream;
+
+/**
+ * Class with config options for Cosmos Client telemetry
+ */
+@Beta(value = Beta.SinceVersion.V4_37_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class CosmosClientTelemetryConfig {
+    private static Logger logger = LoggerFactory.getLogger(CosmosClientTelemetryConfig.class);
+    private static boolean DEFAULT_CLIENT_TELEMETRY_ENABLED = false;
+    private static final Duration DEFAULT_NETWORK_REQUEST_TIMEOUT = Duration.ofSeconds(60);
+    private static final Duration DEFAULT_IDLE_CONNECTION_TIMEOUT = Duration.ofSeconds(60);
+    private static final int DEFAULT_MAX_CONNECTION_POOL_SIZE = 1000;
+    private static EnumSet<TagName> DEFAULT_TAGS = EnumSet.of(
+        TagName.Container,
+        TagName.Operation,
+        TagName.OperationStatusCode,
+        TagName.ClientCorrelationId,
+        TagName.RequestStatusCode,
+        TagName.RequestOperationType,
+        TagName.ServiceAddress
+    );
+
+    private boolean clientTelemetryEnabled;
+    private final Duration httpNetworkRequestTimeout;
+    private final int maxConnectionPoolSize;
+    private final Duration idleHttpConnectionTimeout;
+    private final ProxyOptions proxy;
+    private String clientCorrelationId = null;
+    private EnumSet<TagName> metricTagNames = DEFAULT_TAGS;
+    private MeterRegistry clientMetricRegistry = null;
+    private boolean isClientMetricsEnabled = false;
+    private CosmosClientBuilder owner = null;
```

Comment:
Then we would need to have separate builder for CosmosClientTelemetryConfig and pass that config to builder - which would disallow making this work for client telemetry configs already directly exposed on ClientBuilder - the model I chose here is not just more intuitive (discoverable) but also allows reflecting the client telemetry config changes via two different APIs (avoiding breaking changes) and still capturing all telemetry related configs in one place.
Circular references aren't a problem as long as the whole set of instances with circular references isn't referenced anymore. If no-one holds on to the CosmosClientBuilder anymore GC will collect even with this circular reference. And also - this isn't on the hot path - so even if GC is less efficient - worst case one more collection - it isn't really a concern because CosmosClientBuilder is purely cold path.
So, won't fix IMO.